lib/Optimizer/Optimization.h

```diff
@@ -0,0 +1,8 @@
+using namespace glow;
+
+class Optimization {
+public:
+  Optimization() {}
+  virtual ~Optimization(){};
+  virtual bool run(Function *F) {}
```

Comment:
I'd suggest adding a `getName()` method. It can be very useful for debug printing of pass information and for selecting via the command-line only specific passes to run.